x-pack/plugins/security_solution/public/management/pages/policy/models/advanced_policy_schema.ts

```diff
@@ -0,0 +1,315 @@
+    documentation: '',
+  },
+  {
+    key: 'mac.advanced.kernel.connect',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'mac.advanced.kernel.harden',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'mac.advanced.kernel.process',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'mac.advanced.kernel.filewrite',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'mac.advanced.kernel.network',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.agent.connection_delay',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.global.base_url',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.global.manifest_relative_url',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.global.ca_cert',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.global.public_key',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.global.interval',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.user.base_url',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.user.ca_cert',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.user.public_key',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.artifacts.user.interval',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.elasticsearch.delay',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.elasticsearch.tls.verify_peer',
+    first_supported_version: '7.11',
+    documentation: 'default is true',
+  },
+  {
+    key: 'windows.advanced.elasticsearch.tls.verify_hostname',
+    first_supported_version: '7.11',
+    documentation: 'default is true',
+  },
+  {
+    key: 'windows.advanced.elasticsearch.tls.ca_cert',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.malware.quarantine',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.ransomware.mbr',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.ransomware.canary',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.kernel.connect',
+    first_supported_version: '7.11',
+    documentation: '',
+  },
+  {
+    key: 'windows.advanced.kernel.harden',
```

Comment:
@bjmcnic @willyu-elastic same question

Reply:
I agree this should move out of `kernel` given the direction we're going. My concern with something like `windows.advanced.harden.enabled` would be that what that really refers to is a a parallel technique for protecting our process that will remain useful on Windows 7 and 8.0 if we wish to use it, and while in some ways redundant, would still function even while PPL'd.  Perhaps `windows.advanced.harden.selfprotectprocess` would resolve the ambiguity of simply `enabled`?  @gabriellandau @ferullo @willyu-elastic

Reply:
> that really refers to is a a parallel technique for protecting our process that will remain useful on Windows 7 and 8.0 if we wish to use it

I don't want to support this.  I would, however, like a way to opt out of PPL for debugging/troubleshooting purposes.  Is that feasible?

Reply:
I mean, while anything is possible in software, it'd be significant change to have our install/uninstall logic parse and respect the config.  Additionally, dynamic reconfigs that toggled PPL would force service restarts.  As such, I don't believe having PPL as opt-in/out in the config makes sense.  However, it wouldn't be difficult to allow our executable to install without PPL, or to modify an existing PPL install to un-PPL it and restart it.

In regards to this issue though, are you suggesting we remove the current `harden` option then?  And we could add new options (password protected uninstall, etc.) in a later release?

Reply:
This is carried forward from Endgame.  The Endgame version was [here](https://github.com/elastic/endpoint-dev/blob/8e3fb56911f631f20348636b50b474d615fc5c88/Plugins/BigChief/Lib/endgame/BigChief.cpp#L159-L167).  If we had crashes or hangs before whitelists were applied, we would crash and WER wouldn't be able to dump us.  We couldn't attach a debugger.  Trusted Apps are applied as part of whitelists, so they're ineffective until whitelists are installed.  This feature gave us a kill switch for such situations, though it was cumbersome to enable (modifying files on the SMP and generating new deployment profiles).  Work performed by [this ticket](https://jira.endgames.local/browse/DEF-5143).

`bDisableHardening` is a parameter we [have to pass](https://github.com/elastic/endpoint-dev/blob/201c301a9add5f93bc6955b568ac86c0559e7fd7/Plugins/BigChief/Lib/DriverAuthentication.cpp#L116-L117) when handshaking with the driver.  When I added it for Elastic, hardening was enabled by default.  Rather than hard-coding it to `false`, I exposed it as an opt-out advanced config option like we did for Endgame.  We disabled legacy hardening on Elastic almost immediately thereafter (before that release was made), so it's an unused feature.

> are you suggesting we remove the current harden option then

Yes, unless we clearly and conspicuously indicate that it is an untested and unsupported feature.  I worry some users will [turn it on anyway](https://www.youtube.com/watch?v=hxUMXWtE3qI) and contact us when something doesn't work as expected.

> I mean, while anything is possible in software, it'd be significant change to have our install/uninstall logic parse and respect the config

Understood.  You've put more thought into this than me.  I'm just wondering if there's a way to force the Win7 install behavior on Win8.1+ as a troubleshooting mechanism.  It doesn't have to be through policy, and doesn't have to be toggle-able.  If, for example, we crash on Win8.1 and earlier builds of Win10/Server, WER will encrypt our dumps, making them useless to us.  This doesn't have to be in the initial push.

Reply:
I shouldn't have pinged on this topic here. Let's move this to an endpoint-dev issue discussion so we don't co-opt Jane's PR. This PR is just mirroring the Endpoint's advanced configuration. We'll revisit what is in this document before it is released to make sure it is in sync with the options Endpoint actually supports.

Reply:
thanks all for the feedback. merging this, and we can open a follow-on PR to make sure the schema is exactly right.